src/Modules/homological-algebra.jl

```diff
@@ -548,10 +619,33 @@ julia> matrix(map(Kd, 1))
 ```
 """
 function koszul_complex(V::Vector{T}) where T <: MPolyRingElem
+  r = length(V)
+  iszero(r) && error("list must not be empty")
+  R = parent(first(V))
+  @assert all(x->parent(x)===R, V) "parent mismatch"
+
+  F = FreeMod(R, r)
+  v = sum(x*e for (x, e) in zip(V, gens(F)); init=zero(F))
+  return koszul_complex(v)
+end
+
+function koszul_complex(V::Vector{T}, M::ModuleFP{T}) where T <: MPolyRingElem
+  r = length(V)
+  iszero(r) && error("list must not be empty")
+  R = base_ring(M)
+  @assert all(x->parent(x)===R, V) "parent mismatch"
+
+  F = FreeMod(R, r)
+  v = sum(x*e for (x, e) in zip(V, gens(F)); init=zero(F))
+  return koszul_complex(v, M)
+end
+
+# Deprecated code below; left for reference for how to import things from Singular
+function _koszul_complex_from_singular(V::Vector{T}) where T <: MPolyRingElem
   R  = parent(V[1])
   @assert all(x->parent(x) == R, V)
   n = length(V)
-  KM = [koszul_matrix(V, i) for i=n:-1:1]
+  KM = [_koszul_matrix_from_singular(V, i) for i=n:-1:1]
```

Comment:
@wdecker: I suppose it's because of this line here. The morphisms of the Koszul complex have huge matrices which are sparse. This forces Oscar to allocate the full matrix as a dense matrix. If we had the possibility to pull the morphisms from the Singular side as a list of module elements (images of the generators), we would probably be better off.

Reply:
Even if not urgent since anyway seemingly fast, we should use a ModuleGens to transfer sparse data.